lib/elasticity/index_config.rb

```diff
@@ -14,6 +18,7 @@ def initialize(elasticity_config, defaults = {})
       end
       @elasticity_config = elasticity_config
       yield(self)
+      subclasses_warning
```

Comment:
I think this is a little more complicated than it needs to. We don't need a warning and an error, and ideally the `IndexMapper` class shouldn't need to call methods on the config when creating/recreating the index. If we simply raise the error here, it will prevent the class from being configured and force the user to make the correction rather than catching it in runtime.